plugins/modules/win_scoop.ps1

```diff
@@ -0,0 +1,225 @@
+#!powershell
+
+# Copyright: (c) 2020, Jamie Magee <jamie.magee@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.ArgvParser
+#Requires -Module Ansible.ModuleUtils.CommandUtil
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+  options = @{
+    architecture  = @{ type = "str"; choices = "32bit", "64bit"; aliases = @(, "arch") }
+    independent   = @{type = "bool"; default = $false }
+    global        = @{ type = "bool"; default = $false }
+    name          = @{ type = "list"; elements = "str"; required = $true }
+    no_cache      = @{type = "bool"; default = $false }
+    purge         = @{type = "bool"; default = $false }
+    skip_checksum = @{type = "bool"; default = $false }
+    state         = @{ type = "str"; default = "present"; choices = "present", "absent" }
+  }
```

Comment:
So I see you are using `$module.CheckMode` checks when installing scoop but a module by default does not support check mode. You need to add `supported_check_mode = $true` to `$spec`. When you add it you also need to make sure that the install/uninstall doesn't actually run in check mode, or add a `-whatif` like flag to scoop if it supports it.